R/input-submit.R

```diff
@@ -0,0 +1,161 @@
+#' Create a textarea input control with explicit submission
+#'
+#' Creates a textarea input where users can enter multi-line text and submit
+#' their input using a dedicated button or keyboard shortcut. This control is
+#' ideal when you want to capture finalized input, rather than reacting to every
+#' keystroke, making it useful for chat boxes, comments, or other scenarios
+#' where users may compose and review their text before submitting.
+#'
+#' @param id The input ID.
+#' @param label The label to display above the input control. If `NULL`, no
+#'   label is displayed.
+#' @param placeholder The placeholder text to display when the input is empty.
+#'   This can be used to provide a hint or example of the expected input.
+#' @param value The initial input text. Note that, unlike [textAreaInput()],
+#'   this won't set a server-side value until the value is explicitly submitted.
+#' @param ... Currently not used.
+#' @param button A [tags] element to use for the submit button. It's recommended
+#'   that this be a [input_task_button()] since it will automatically provide a
+#'   busy indicator (and disable) until the next flush occurs. Note also that if
+#'   the submit button launches a [ExtendedTask], this button can also be bound
+#'   to the task ([bind_task_button()]) and/or manually updated for more
+#'   accurate progress reporting ([update_task_button()]).
+#' @param width Any valid CSS unit (e.g., `width="100%"`).
+#' @param submit_key A character string indicating what keyboard event should
+#'   trigger the submit button. The default is `enter`, which will submit the
+#'   input when the user presses the Enter/Return key. The `enter+modifier`
+#'   option will submit the input when the user presses the Enter key while
+#'   holding down Ctrl/Cmd.
+#'
+#' @return A textarea input control that can be added to a UI definition.
+#'
+#' @seealso [update_submit_textarea()], [input_task_button()]
+#'
+#' @examplesIf rlang::is_interactive()
+#'
+#' ui <- page_fluid(
+#'   input_submit_textarea("text", placeholder = "Enter some input..."),
+#'   verbatimTextOutput("value")
+#' )
+#' server <- function(input, output) {
+#'   output$value <- renderText({
+#'     req(input$text)
+#'     Sys.sleep(2)
+#'     paste("You entered:", input$text)
+#'   })
+#' }
+#' shinyApp(ui, server)
+#'
+#' @section Server value:
+#' A character string of the text input. The default value is `""` even if
+#' `value` is provided. The value will only be set/updated when the user submits
+#' the input by pressing the Enter key or clicking the submit button.
```

Comment:
Small wording adjustments because I was initially confused by "the default value is `""`".

```suggestion
#' @section Server value:
#' The server receives a character string containing the user's text input.
#'
#' **Important:** The initial server value is always `""` (empty string),
#' regardless of any `value` parameter provided to `input_submit_textarea()`.
#' The server value updates only when the user explicitly submits the input by
#' either pressing the Enter key (possibly with a modifier key) or clicking the
#' submit button.
```